lib/src/scenario_generator.dart

```diff
@@ -13,18 +13,25 @@ void parseScenario(
 ) {
   sb.writeln(
       '    $testMethodName(\'\'\'$scenarioTitle\'\'\', (tester) async {');
+  sb.writeln('      try {');
```

Comment:
Add this `try..finally` bloc only when `hasTearDown` is `true`. The simplicity and minimality of generated code are essential. We generate only the code that the developer would have to write manually, if you don't have anything to put into the `finally` bloc - you would not add it.